crates/rustc_codegen_spirv/src/spirv_type_constraints.rs

```diff
@@ -745,7 +745,7 @@ pub fn instruction_signatures(op: Op) -> Option<&'static [InstSig<'static>]> {
         // SPV_AMD_shader_fragment_mask
         Op::FragmentMaskFetchAMD | Op::FragmentFetchAMD => reserved!(SPV_AMD_shader_fragment_mask),
         // SPV_KHR_shader_clock
-        Op::ReadClockKHR => reserved!(SPV_KHR_shader_clock),
+        Op::ReadClockKHR => {}
```

Comment:
ping @eddyb to review this bit, I forget how spirv_type_constraints works

Reply:
Looking at [the extension](https://github.com/KhronosGroup/SPIRV-Registry/blob/004f312872a014d6b7cf33e33fbd235c33a6be27/extensions/KHR/SPV_KHR_shader_clock.asciidoc#instructions), looks like the instruction doesn't have multiple types that are related, so nothing to here.

However, I'd suggest this change:
```suggestion
        Op::ReadClockKHR => {
            // NOTE(eddyb) we actually use these despite not being in the standard yet.
            // reserved!(SPV_KHR_shader_clock)
        }
```

We're ending up with a lot of these, I wish the standard contained the current status of the extensions instead of just describing the instructions as "reserved" (which is why I initially avoided handling them here).